app/models/permitted_params/allowed_settings.rb

```diff
@@ -29,6 +29,31 @@ def all
       keys
     end
 
+    def scalar_and_complex_filters
+      restricted_keys = Set.new(self.restricted_keys)
+      scalar_filters = []
+      complex_filters = {}
+      Settings::Definition.all.each do |key, definition| # rubocop:disable Rails/FindEach
+        next if restricted_keys.include?(key)
+
+        case definition.format
+        when :hash
+          complex_filters[key] = {}
+        when :array
+          complex_filters[key] = []
+        else
+          scalar_filters << key
+        end
```

Comment:
From the docs, the only argument for [`ActionController::Parameters#permit`](https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-permit) is a list of filters, no kwargs. So we would not need to seperate scalar and complex filters and could simply rewrite this to:

```suggestion
        case definition.format
        when :hash
          scalar_filters << { key => {} }
        when :array
          scalar_filters << { key => [] }
        else
          scalar_filters << key
        end
```